client/sites-dashboard/hooks/use-check-site-transfer-status.tsx

```diff
@@ -0,0 +1,53 @@
+import { useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchAutomatedTransferStatus } from 'calypso/state/automated-transfer/actions';
+import { transferStates } from 'calypso/state/automated-transfer/constants';
+import {
+	getAutomatedTransferStatus,
+	isFetchingAutomatedTransferStatus,
+} from 'calypso/state/automated-transfer/selectors';
+
+interface SiteTransferStatusProps {
+	siteId: number | null;
+	intervalTime?: number;
+}
+
+export const useCheckSiteTransferStatus = ( {
+	siteId,
+	intervalTime = 3000,
+}: SiteTransferStatusProps ) => {
+	const dispatch = useDispatch();
+
+	const transferStatus = useSelector( ( state ) => getAutomatedTransferStatus( state, siteId ) );
+	const isFetchingTransferStatus = useSelector( ( state ) =>
+		isFetchingAutomatedTransferStatus( state, siteId )
+	);
+
+	const isTransferCompleted = transferStatus === transferStates.COMPLETE;
+	const isTransferring =
+		transferStatus !== null && transferStatus !== transferStates.NONE && ! isTransferCompleted;
```

Comment:
We should check for `.FAILURE` and `.ERROR` here too.

Reply:
yeah we should check for failures. This currently doesn't work on the list view, should we do the list view also like the plan expired? A column shows the expire notice.

Reply:
That's something that we need to figure out with @javierarce but I'm fine with the list view being the same as the plan expired, except with a spinner included.